### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by tshemsedinov**

### CODE REVIEW

File: tools/Resizer/resizer.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+            # additional row number
+            for rn in range(coef):
+                # add new void row
+                resized_result.append([])
+                for j in range(self.width):
+                    # additional column number
+                    for cn in range(coef):
+                        # add new void pixel
+                        resized_result[-1].append([])
+                        for rgb in range(3):
```
Comment (tshemsedinov) [inline tools/Resizer/resizer.py]: Terrible nesting

File: tools/Resizer/resizer.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+                        # copy pixel from original image
+                        resized_result[-1].append(self.matrix[i][j])
+
+        self.file_size = self.file_size\
+            + 3*(self.width*self.height)*(coef**2 - 1)
+        self.width = int(self.width * coef)
+        self.height = int(self.height * coef)
+        self.matrix = resized_result
+
+    def size_up_matrix_bilinear(self, coef):
```
Comment (tshemsedinov) [inline tools/Resizer/resizer.py:182]: Long method, decompose

File: tools/Resizer/resizer.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        # Height (22-25)
+        write_bytes_from_int(22, 4, self.height, destination)
+        # Planes (26-27)
+        write_bytes_from_int(26, 2, self.planes, destination)
+        # Bits Per Pixel (28-29)
+        write_bytes_from_int(28, 2, self.bits_per_pixel, destination)
+        # Compression (30-33)
+        write_bytes_from_int(30, 4, self.compression, destination)
+        # (compressed) ImageSize (34-37), defaults to 0
+        write_bytes_from_int(34, 4, 0, destination)
```
Comment (tshemsedinov) [inline tools/Resizer/resizer.py]: Magic numbers here and everywhere
AmbitiousPracticalJabiru-small

File: tools/Resizer/resizer.py [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        for i in range(self.height):
+            # for additional row number
+            for rn in range(coef):
+                # add new void row
+                resized_result.append([])
+                for j in range(self.width):
+                    # for additional column number
+                    for cn in range(coef):
+                        # copy pixel from original image
+                        resized_result[-1].append(self.matrix[i][j])
```
Comment (tshemsedinov) [inline tools/Resizer/resizer.py]: Nested loops

**Review by tshemsedinov — COMMENTED** [review]
All bad practices are here
- long methods
- nested loops
- bad naming
- magic numbers
- obvious comments

File: tools/Resizer/resizer.py (hunk 3 above)
Comment (cyberlord-coder-228) [reply tools/Resizer/resizer.py]: This is not a magic numbers, they are used in the bmp file format, which are not going to change soon ._.)
Any way to remove them will bring just move confusion as what they are reffering to

Comment (Nocommas555) [reply tools/Resizer/resizer.py]: http://www.ece.ualberta.ca/~elliott/ee552/studentAppNotes/2003_w/misc/bmp_file_format/bmp_file_format.htm

Comment (Nocommas555) [reply tools/Resizer/resizer.py]: the fields are described above the function that writes to them

**Comment by Nocommas555** [comment]
we are ready to present our work

**Comment by Nocommas555** [comment]
when will you be available to set up a video conference for us?

**Comment by Nocommas555** [comment]
don't mind the doubled commits

**Comment by Nocommas555** [comment]
I tried to resign all my commits, but they still did not show up on my profile

**Pull request merged**